spring-cloud-netflix-eureka-client/src/main/java/org/springframework/cloud/netflix/eureka/EurekaHealthCheckHandler.java

```diff
@@ -104,20 +115,39 @@ void populateHealthIndicators(Map<String, HealthIndicator> healthIndicators) {
 		}
 	}
 
+	void populateReactiveHealthIndicators(
+			Map<String, ReactiveHealthIndicator> reactiveHealthIndicators) {
+		for (Map.Entry<String, ReactiveHealthIndicator> entry : reactiveHealthIndicators
+				.entrySet()) {
+			this.reactiveHealthIndicators.put(entry.getKey(), entry.getValue());
+		}
+	}
+
 	@Override
 	public InstanceStatus getStatus(InstanceStatus instanceStatus) {
 		return getHealthStatus();
 	}
 
 	protected InstanceStatus getHealthStatus() {
-		Status status = getStatus(statusAggregator);
+		Status status = getStatus(this.statusAggregator);
 		return mapToInstanceStatus(status);
 	}
 
 	protected Status getStatus(StatusAggregator statusAggregator) {
 		Status status;
-		Set<Status> statusSet = healthIndicators.values().stream().map(HealthIndicator::health).map(Health::getStatus)
-				.collect(Collectors.toSet());
+		Set<Status> statusSet = new HashSet<>();
+		if (this.healthIndicators != null) {
+			statusSet.addAll(
+					this.healthIndicators.values().stream().map(HealthIndicator::health)
+							.map(Health::getStatus).collect(Collectors.toSet()));
+		}
+
+		if (this.reactiveHealthIndicators != null) {
+			statusSet.addAll(this.reactiveHealthIndicators.values().stream()
+					.map(ReactiveHealthIndicator::health).map(health -> {
+						return health.block().getStatus();
```

Comment:
I would usually suggest using non-blocking calls on reactor objects, but since the entire `EurekaHealthCheckHandler` is not reactive, I think we can leave it this way.